core/io/resource_loader.cpp

```diff
@@ -999,6 +999,9 @@ void ResourceLoader::load_translation_remaps() {
 
 void ResourceLoader::clear_translation_remaps() {
 	translation_remaps.clear();
+	while (unlikely(remapped_list.first())) {
+		remapped_list.remove(remapped_list.first());
+	}
```

Comment:
I'm not sure this is the correct fix. Is it really `unlikely()` that there would be leftover remaps? Why does it happen then?
I don't see any code that actually removes items from `remapped_list` apart from
```
core/resource.cpp
327:            ResourceLoader::remapped_list.remove(&remapped_list);
```

Reply:
I'm not sure about `unlikely`, comparing with `NULL`/`nullptr` is probably the better way. 

But `remapped_list` get never cleared, so if I have any translation remaps in the project, I end with SelfList leak (this happens in OSX as well as iOS for me). 

Also, removing from `remapped_list` could be done in `Resource::~Resource` for preventing other runtime leaks, but I need more time diving deep into `Resource`.

Reply:
btw, current code completely clears `remapped_list`

Reply:
@reduz suggests that just adding `remapped_list.clear();` instead of this while loop might be the proper fix.

Reply:
I'd like to, but `SelfList` has no `clear` method. I can implement it, is this the way?

Reply:
I guess it's fine as is then, I'd just remove `unlikely()` as it might be counter-productive (`unlikely()` is used to optimize the codepath assuming that it will be `false` most of the time - but in the case where it's actually `true`, we'd lose some performance).